### ISSUE TITLE
[bug] estimatefee & estimatesmartfee always return -1 as result

### ISSUE THREAD

**Issue by mangekyousharingan** [dogecoin/dogecoin#3385]
Hi!

Want to report that whenever I call DOGE node, the result is always -1 on both estimatefee and estimatesmartfee RPC methods.

Is there any issue related to this? 

example call:

```
curl --location 'https://rpc.coinsdo.net/doge' \
--header 'Content-Type: application/json' \
--data '{"jsonrpc": "1.0", "id": "1", "method": "estimatefee", "params": [6]}'
```

response:

```
{
    "result": -1,
    "error": null,
    "id": "1"
}
```

**Comment by patricklodder** [comment]
I've been able to reproduce this right now (although until yesterday it was working fine) so current blocks are triggering a condition to invalidate the result - I'm not sure which one yet.

**Comment by mangekyousharingan** [comment]
@patricklodder thanks for sharing it. I guess we have to wait for proper fix and estimate fee on our own.

**Comment by patricklodder** [comment]
I'm working on a patch, and have done minimal orientation to manage an expedited release for it.

There are two major problems with the fee estimation logic:

1. ***The mempool is highly volatile***, especially because the past couple of days people have been scripting and racing each other with intrinsically economically unsound transactions. The perceived external value makes people spend over thousands times the DOGE they are transacting. **This can be fixed but it needs to be carefully tuned, or everyone, including people that do economically meaningful transactions, will overpay significantly** (see below for an illustration.) Last year I did a lot of research work to enable the in-progress 1.21 major release with a good baseline, but that was before this behavior occurred and now I have to redo the tuning part to take into account "economic insanity", which takes time.
2. ***Blockspace is less reliably used in Dogecoin***. Unlike Bitcoin, Dogecoin does not try to prescribe a particular miner block templating behavior. This is because there is no mempool level consensus (also not on Bitcoin) so it makes no sense to have illusions of safety, and the economic incentive for mining Dogecoin blocks is not to extract as much fee as possible, because nowadays its subsidy is higher than all Dogecoin fees and all Litecoin subsidy AND fees combined: a miner has a bigger incentive to not have orphaned blocks than to include every transaction.

Over the past few hours I've seen the below estimates on a tuned, reconfigured model, but in reality your transaction will get mined with 0.001 DOGE/kb within 5 blocks with 70% certainty. So this already improved model is still overestimating fee 541 times because of the fee race happening, even though the size of a default configured mempool is < 2 blocks worth of transactions at the moment.

```
2 block: 0.78259712
3 block: 0.54144695
5 block: 0.54144695
7 block: 0.48850724
10 block: 0.40863303
15 block: 0.40863303
20 block: 0.40863303
25 block: 0.3433009
```

**Comment by patricklodder** [comment]
I'm going to propose a series of pull requests to address this issue over the coming week, starting tonight:

- [ ] A new default configuration that makes `estimatefee` and `estimatesmartfee` functional, also in periods of spam.
- [ ] A facility for users to configure this functionality to their liking, because developers should not dictate the fee a user pays.
- [ ] Expansive documentation to using smartfees, because using this facility (or any other fee estimation facility) does not come without caveats and risk.

This basically changes the automatic fee estimation feature from being unmaintained and a depreciation candidate, to an actively supported feature, and that is a policy change for Dogecoin Core (with 1.14.0, smartfees was said to no longer be supported.) Because there is plenty of time needed for testers to get results, consider this an open invitation for anyone to opine in on the desirability of supporting this feature. 

**Please comment your thoughts.**

**Comment by DogestMaximus** [comment]
Thank you for this

**Comment by RnoHach** [comment]
So in essence, this will bring smart fees back?

I don’t understand the technicalities, but I do care, so please do not bite my head off. Maybe someone who doesn’t mind could explain it to me like I’m five? 😅

**Comment by patricklodder** [comment]
> So in essence, this will bring smart fees back?

Yes, but note that for now, the scope of my proposal to bring back the API method into active support, because that's what this issue is about. If that is successful, we can consider re-hanging it into the wallet as an optional or, if we're confident that it won't make everyone overpay, even the default mechanism to determine fee.

**Comment by mangekyousharingan** [comment]
`estimatefee` back to be working as it was ;) I guess it got fixed by itself? 🤔 Thank you @patricklodder for all the input! 🙏

**Comment by patricklodder** [comment]
I'm running clean 1.14.6 and #3389 nodes for monitoring / comparison, and there were 3 moments today where this happened again for me (time in UTC):

- From 05:39 to 07:56 (2h17m)
- From 10:25 to 11:10 (45m)
- From 11:11 to 11:12 (1m)

Screenshot from my prometheus/grafana that monitors these nodes:

<img width="1351" alt="12h estimatefee availability for 1.14.6" src="https://github.com/dogecoin/dogecoin/assets/1410115/8f9aabd9-b040-4806-9ffd-8605425975b2">

**Comment by victorsk2019** [comment]
> I'm running clean 1.14.6 and #3389 nodes for monitoring / comparison, and there were 3 moments today where this happened again for me (time in UTC):

Thank you for sharing this.  I also installed modified /src/policy/fees.h from #3389 as a patch to test on my full node and `estimatesmartfee` was returning fee every time for various block parameter values, even when transaction count in mempool was around 50-60K.

**Comment by mwbro1** [comment]
hi 
I have same issue when trying to send Doge i get message "insufficient funds" no matter the amount. 
![doge](https://github.com/dogecoin/dogecoin/assets/153525064/24112219-9432-4f04-93fb-ea5d20e1786f)

**Comment by danielw86dev** [comment]
Hej @mangekyousharingan :) nie chcesz zrobić review tłumaczenia? https://github.com/dogecoin/dogecoin/pull/3431

**Comment by patricklodder** [comment]
This was fixed with #3389 and planned to be released next week with 1.14.7.

I have made a new tracker at #3440 for the whole chain of fixes needed to get this functionality back into shape.
